cmd/authelia/main.go

```diff
@@ -27,6 +27,13 @@ var configPathFlag string
 //nolint:gocyclo // TODO: Consider refactoring/simplifying, time permitting.
 func startServer() {
 	logger := logging.Logger()
+	created, err := configuration.GenerateIfNotFound(configPathFlag)
```

Comment:
I don't understand how this relates to the problem mentioned in the title of the PR. Should probably be part of a dedicated PR where we want to generate a default configuration if none is provided.

It's a good idea to generate the configuration but I would rather let Authelia fails if the file is not found and rather provide this feature using a new documented CLI command. I'd rather make it explicit that something is wrong when starting up Authelia rather than let the user figure out very late that the config was not at the right place and another one has been generated "silently" in the meantime (even a log line saying the config has been generated means silent for me because I make the assumption not everyone read logs carefully and I wan't to avoid that toil).

Reply:
How it relates is if a user tries to hash-password and gives an invalid path it creates a configuration file there. This behavior is used with the container on the config volume on startup.

Reply:
Perhaps it makes sense to handle these things in separate PRs - close this and create a new one once the configuration issue has been resolved in a new PR? That way we can easily focus on getting each part done in a way that works for everyone.

Reply:
Yeah that makes sense to me @AlexGustafsson If you just do the basic portion of getting the hash-password to use the config if provided I will do that work. I'm actually doing work to implement a new config lib which should make all of this a lot cleaner.

Reply:
@james-d-elliott could you ping me here once that's merged or ready for merge? I'll fix it then.

Reply:
Absolutely! Should be done this weekend, but will ping you. It's a fairly hefty refactor of configuration, but I can help merge the changes if you have issues either way.